action-binding-generator/src/main/kotlin/io/github/typesafegithub/workflows/actionbindinggenerator/generation/Generation.kt

```diff
@@ -403,19 +488,74 @@ private fun Metadata.buildCommonConstructorParameters(
                 ).build(),
         )
 
-private fun ParameterSpec.Builder.defaultValueIfNullable(input: Input): ParameterSpec.Builder {
-    if (!input.shouldBeNonNullInBinding()) {
+private fun ParameterSpec.Builder.defaultValueIfNullable(
+    input: Input,
+    untypedClass: Boolean,
+    typedInput: Boolean,
+): ParameterSpec.Builder {
+    if (input.shouldBeNullable(untypedClass, typedInput)) {
         defaultValue("null")
     }
     return this
 }
 
+private fun TypeSpec.Builder.addInitializerBlockIfNecessary(
+    metadata: Metadata,
+    inputTypings: Map<String, Typing>,
+    untypedClass: Boolean,
+): TypeSpec.Builder {
+    if (untypedClass || metadata.inputs.isEmpty() || metadata.inputs.none { inputTypings.containsKey(it.key) }) {
+        return this
+    }
+    addInitializerBlock(metadata.initializerBlock(inputTypings))
+    return this
+}
+
+private fun Metadata.initializerBlock(inputTypings: Map<String, Typing>): CodeBlock {
+    val codeBlockBuilder = CodeBlock.builder()
+    var first = true
+    inputs
+        .filter { inputTypings.containsKey(it.key) }
+        .forEach { (key, input) ->
+            if (!first) {
+                codeBlockBuilder.add("\n")
+            }
+            first = false
+            val propertyName = key.toCamelCase()
+            codeBlockBuilder
+                .add(
+                    """
+                    require(!((%1N != null) && (%1L_Untyped != null))) {
+                        %2S
+                    }
+
+                    """.trimIndent(),
+                    propertyName,
+                    "Only $propertyName or ${propertyName}_Untyped must be set, but not both",
+                )
+            if (input.shouldBeRequiredInBinding()) {
+                codeBlockBuilder
+                    .add(
+                        """
+                        require((%1N != null) || (%1L_Untyped != null)) {
+                            %2S
+                        }
+
+                        """.trimIndent(),
+                        propertyName,
+                        "$propertyName or ${propertyName}_Untyped must be set, one of them is required",
```

Comment:
```suggestion
                        "Either $propertyName or ${propertyName}_Untyped must be set, one of them is required",
```
to emphasize that it's like logical xor, not or?